components/avatar/Avatar.tsx

```diff
@@ -0,0 +1,120 @@
+/**
+ * Author: bladimirardiles
+ * Component File Name: Avatar.js
+ * Component Name: Avatar
+ * Project: design-system
+ * Date: Sat 09/12/2023 - 11:34
+ */
+
+import React, { useId, useMemo } from 'react';
+import { IAvatarComponentProps } from './avatar.interface';
+import Button from '../button/Button';
+import { EIConTypes } from '../icon/icon.enum';
+import Image from '../image/Image';
+import { EImageLoadingType } from '../image/image.enum';
+import { get as _get } from 'lodash';
+import { File } from 'buffer';
+import { EAvatarSizes } from './avata.enum';
+import {
+  editableLabel,
+  topContainerLg,
+  topContainerMd,
+  topContainerSm,
+  topContainerXl,
+} from './avatar-css';
+
+const Avatar = ({
+  imgSrc,
+  onError,
+  onSuccess,
+  name,
+  editable,
+  size = EAvatarSizes.SM,
+}: IAvatarComponentProps) => {
+  // const cssClasses = modifierState(componentName, [], []);
+
+  const topContainerClasses = useMemo(() => {
+    if (size === EAvatarSizes.XL || editable) {
+      return topContainerXl.join(' ');
+    } else if (size === EAvatarSizes.LG) {
+      return topContainerLg.join(' ');
+    } else if (size === EAvatarSizes.MD) {
+      return topContainerMd.join(' ');
+    } else if (size === EAvatarSizes.SM) {
+      return topContainerSm.join(' ');
+    }
+  }, [size, editable]);
+
+  const identifier = useId();
+  return (
+    <div className="c-avatar relative">
+      <div className={topContainerClasses}>
+        {imgSrc && (
+          <Image
+            src={imgSrc}
+            alt={name ? name : ''}
+            lazy={EImageLoadingType.EAGER}
+          />
+        )}
+      </div>
+      {editable && (
+        <div className="absolute bottom-0 right-0">
+          <input
+            style={{ display: 'none' }}
+            id={identifier}
+            accept="image/*"
+            className=""
+            onChange={(evt) => {
+              try {
+                let imageFile: File | null = _get(
+                  evt,
+                  'currentTarget.files[0]',
+                  null,
+                );
+                if (imageFile) {
+                  imageFile = imageFile as File;
+                  // @ts-expect-error filereader is not recognized ¯\_(ツ)_/¯
+                  const reader = new FileReader();
```

Comment:
why do you need a file reader in this component?

Reply:
Shouldn't we implemented the file upload in a funtion within the Web App and provide only an onClick on the component itself? what do you think about it?

Reply:
I think the avata itself should pass the prepared file to the parent component. no more than that. thats why we need the fileReader. Also this way we could already update the image with the buffer. Letting us tho show a confirm and by YES, then we proceed to save the avatar. 

We can see how the approach goes while implementing and change it